lib/shopify_api/resources/base.rb

```diff
@@ -28,6 +28,32 @@ def as_json(options = nil)
     end
 
     class << self
+      def search(params = {})
+        find(:all, params: params)
+      end
+
+      def auto_paging_each(opts)
+        opts[:limit] ||= 50
+        page_limit = opts[:limit]
+
+        total_count = self.count(opts)
+        pages = total_count/page_limit + 1
+        current_page = opts[:page] || 0
+        records = []
+
+        while current_page != pages
+          self.search(opts.merge(page: current_page)).each do |record|
+            records << record
+
+            yield(record) if block_given?
+          end
+
+          current_page += 1
+        end
+
+        records
+      end
```

Comment:
I usually write this to keep incrementing the page until no records are returned.